qiskit/validation/base.py

```diff
@@ -106,20 +109,103 @@ def __call__(self, model_cls):
         self._schema_cls.model_cls = model_cls
         model_cls.schema = self._schema_cls()
-        model_cls._validate = self._validate
+
+        # Append the methods to the Model class.
         model_cls.to_dict = self._to_dict
         model_cls.from_dict = classmethod(self._from_dict)
+        model_cls._validate = self._validate
         model_cls.__init__ = self._validate_after_init(model_cls.__init__)
+
+        # Add a Schema that performs minimal validation to the Model.
+        model_cls.shallow_schema = self._create_shallow_schema(self._schema_cls)
+
         return model_cls
 
+    def _create_shallow_schema(self, schema_cls):
+        """Create a Schema with minimal validation for compound types.
+
+
+        This is a helper for performing the initial validation when
+        instantiating the Model via **kwargs. It works on the assumption that
+        **kwargs will contain:
+        * for compound types (`Nested`, `BasePolyField`), it will already
+          contain `BaseModels`, which should have been validated earlier
+          (during _their_ instantiation), and only type checking is performed.
+        * for `Number` and `String` types, both serialized and deserialized
+          are equivalent, and the shallow_schema will try to serialize in
+          order to perform stronger validation.
+        * for the rest of fields (the ones where the serialized and deserialized
+          data is different), it will contain _deserialized_ types that are
+          passed through.
+
+        The underlying idea is to be able to perform validation (in the schema)
+        at only the first level of the object, and at the same time take
+        advantage of validation during **kwargs instantiation as much as
+        possible (mimicking `.from_dict()` in that respect).
+
+        Returns:
+            BaseSchema: a copy of the original Schema, overriding the
+                ``_deserialize()`` call of its fields.
+        """
+        shallow_schema = schema_cls()
+        for _, field in shallow_schema.fields.items():
+            if isinstance(field, fields.Nested):
+                field._deserialize = partial(self._overridden_nested_deserialize, field)
+            elif isinstance(field, BasePolyField):
+                field._deserialize = partial(self._overridden_basepolyfield_deserialize, field)
+            elif not isinstance(field, (fields.Number, fields.String)):
+                field._deserialize = partial(self._overridden_field_deserialize, field)
+        return shallow_schema
+
+    @staticmethod
+    def _overridden_nested_deserialize(instance, value, _, data):
+        """Helper for minimal validation of fields.Nested."""
+        if instance.many and not is_collection(value):
+            instance.fail('type', input=value, type=value.__class__.__name__)
+
+        if not instance.many:
+            value = [value]
+
+        for v in value:
+            if not isinstance(v, instance.schema.model_cls):
```

Comment:
Hmm, I think that would lead to false positives (for example. a `Nested(FooSchema)` would _not_ raise here if it encounters a `BarModel` - let's see how it goes without it!